### ISSUE TITLE
 the direct positioning got stuck

### ISSUE THREAD

**Issue by ming1j2** [rsasaki0109/lidar_localization_ros2#35]
Hello, I would like to ask why I entered a source point cloud and opened the real-time point cloud of the radar. However, during operation, there was a lag phenomenon during real-time radar positioning, and in the end, the direct positioning got stuck

**Comment by rsasaki0109** [comment]
Sorry, I just noticed your comment.
I would like to know more details. For example, whether the environment is indoors or outdoors, or if you have done any parameter tuning for that.
Also, I would like the YAML file. It would be better if there are images or videos of the position estimation.
Setting ndt_num_threads to 0 (meaning maximum allowable threads are used) can sometimes cause problems, so it might be good to adjust it.

**Comment by ming1j2** [comment]
In the indoor environment, my yaml file only changed the input path of the source point cloud for map_path and the number of ndtnum_threads to 4. Other changes may have been made to the point cloud topic and imu topic. The effect is that after opening it, it can be repositioned and registered, but after a period of time, it will occasionally get stuck, and in the end, the positioning will just get stuck. I don't know if there was a problem with the change

**Comment by ming1j2** [comment]
And there is also a strange issue, which is that after changing the number of ndtnum_threads to 4, but the info display result of ROS2 is still 0, which makes me not sure if the subsequent repositioning has undergone multi-threaded acceleration

**Comment by rsasaki0109** [comment]
In ROS2, you need to rebuild to apply parameter changes.
If you add the --symlink-install option during the build, you no longer need to rebuild.

**Comment by ming1j2** [comment]
But what is the problem with this repositioning getting stuck in the end? Did you encounter this issue while trying out the indoor scene radar

**Comment by rsasaki0109** [comment]
I'm not quite sure what the problem is, so I'd like to see an image or video of rviz.

**Comment by ming1j2** [comment]
![Uploading 微信图片_20240417130219.jpg…]()

**Comment by ming1j2** [comment]
![Uploading 1.jpg…]()

**Comment by ming1j2** [comment]
![Uploading 1.jpg…]()

**Comment by ming1j2** [comment]
![截图 2024-04-17 13-27-25](https://github.com/rsasaki0109/lidar_localization_ros2/assets/166522594/9ab0d183-6e58-4e4d-a6bf-1e5022152bd2)

**Comment by ming1j2** [comment]
Just like this

**Comment by rsasaki0109** [comment]
Can you upload the rviz screen? I'd like to see the map, trajectory, scans, etc.

**Comment by rsasaki0109** [comment]
The delta angle is large.
By the way, have you been swinging the LiDAR around by hand?
The current LiDAR localization uses IMU coupling based on LeGO-LOAM which might be quite sensitive to that.

**Comment by ming1j2** [comment]
Yes, I did swing the LiDAR with my hand. Can't I swing it with my hand? If rviz is turned on, then the trajectory is the one I swing the radar with my hand

**Comment by rsasaki0109** [comment]
The stuck-positioning case (often scan matching failing under fast hand-held motion without IMU) is covered in the stuck positioning section of [docs/troubleshooting.md](https://github.com/rsasaki0109/lidar_localization_ros2/blob/main/docs/troubleshooting.md), with `/alignment_status` interpretation to identify it. Closing as documented.
